scripts/feature_tests/cleanup_env.sh

```diff
@@ -0,0 +1,73 @@
+#!/bin/bash
+set -xe
+
+# shellcheck disable=SC1091
+source lib/logging.sh
+# shellcheck disable=SC1091
+source lib/common.sh
+# shellcheck disable=SC1091
+source lib/network.sh
+
+export GOPATH="/home/${USER}/go"
+M3PATH="${M3PATH:-${GOPATH}/src/github.com/metal3-io}"
+BMOPATH="${BMOPATH:-${M3PATH}/baremetal-operator}"
+CAPIPATH="${CAPIPATH:-${M3PATH}/cluster-api}"
+CAPM3_BASE_URL="${CAPM3_BASE_URL:-metal3-io/cluster-api-provider-metal3}"
+CAPM3RELEASEPATH="${CAPM3RELEASEPATH:-https://api.github.com/repos/${CAPM3_BASE_URL}/releases/latest}"
+CAPM3RELEASE="${CAPM3RELEASE:-$(get_latest_release "${CAPM3RELEASEPATH}")}"
+NAMESPACE="metal3"
+
+# Remove BMH
+pushd "${BMOPATH}"
+kubectl delete -f bmhosts_crs.yaml -n "${NAMESPACE}"
+popd
+
+# Kill and remove the running ironic containers
+for name in ironic ironic-inspector ironic-endpoint-keepalived dnsmasq httpd mariadb; do
+  sudo "${CONTAINER_RUNTIME}" ps | grep -w "$name$" && sudo "${CONTAINER_RUNTIME}" kill $name
+  sudo "${CONTAINER_RUNTIME}" ps --all | grep -w "$name$" && sudo "${CONTAINER_RUNTIME}" rm $name -f
+done
+
+pushd "${CAPIPATH}"
+cd bin
+./clusterctl delete --namespace metal3 --all -v5
+popd
+
+declare -a OBJECTS=("cluster" \
+"baremetalhost" \
+"kubeadmconfig" \
+"kubeadmcontrolplane" \
+"machines" \
+"metal3cluster" \
+"metal3machine" \
+"metal3machinetemplate" \
+)
+
+delete_finalizers(){
+shift
+local kind
+for name in "${OBJECTS[@]}"; do
+  kind="$(kubectl get "${name}" -n "${NAMESPACE}" -o name)"
+  for item in "${kind[@]}"; do
+          kubectl delete "${item}" -n "${NAMESPACE}"
+          kubectl patch "${item}" -n "${NAMESPACE}" -p '{"metadata":{"finalizers":[]}}' --type=merge
+  done
+  #process_status $?
+done
+}
+
+delete_finalizers
+
+# Re-create ironic containers and BMH
+pushd "${BMOPATH}"
+./tools/run_local_ironic.sh
+popd
+
+pushd "${CAPIPATH}"
+cd bin
+./clusterctl init --infrastructure=metal3:"${CAPM3RELEASE}"  --target-namespace metal3 --watching-namespace metal3 -v5
+popd
+
+pushd "${BMOPATH}"
+kubectl apply -f bmhosts_crs.yaml -n "${NAMESPACE}"
+popd
```

Comment:
Should this wait for the bmh to move to ready state ?

Reply:
you could actually call the verification script here 04_*

Reply:
> you could actually call the verification script here 04_*

good point.

Reply:
> Should this wait for the bmh to move to ready state ?

I think that will be taken care by 04_* script if we add it at the end of this script.